src/main/java/Grid.java

```diff
@@ -60,9 +60,10 @@ public void clearGrid() {
     public void getNextGeneration() {
         Square[][] newBoard = new Square[WIDTH][HEIGHT];
         aliveSquares.clear();
+        deadSquares.clear();
 
-        for (int y = 0; y < HEIGHT; y++) {
-            for (int x = 0; x < WIDTH; x++) {
+        for (int y = 0; y < newBoard.length; y++) {
+            for (int x = 0; x < newBoard[y].length; x++) {
```

Comment:
why would newBoard.length be different than newBoard[y].length, since it's going until whatever y reached?

Reply:
In our specific board it's not different because our board is a square. newBoard[y] gets the length of the array at index y

Reply:
https://www.devcubicle.com/java-2d-array-length/